### PR TITLE
Add documentation for THM kernels and scalar kernels

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #19737

To be considered after the previous one has been merged
EDIT: other one merged

### CODE REVIEW

File: modules/thermal_hydraulics/doc/content/source/base/FlowModelSinglePhase.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# FlowModelSinglePhase
+
+The `FlowModelSinglePhase` creates for [FlowChannel1Phase.md] components using the three equation model:
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/base/FlowModelSinglePhase.md:3]:
```suggestion
The `FlowModelSinglePhase` creates for [FlowChannel1Phase.md] components using the three equations model:
```

File: modules/thermal_hydraulics/doc/content/source/kernels/ADHeatConductionRZ.md [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
-MooseObject; however, what is contained is ultimately determined by what is necessary to make the
-documentation clear for users.
-
 !syntax description /Kernels/ADHeatConductionRZ
 
-## Overview
-
-!! Replace these lines with information regarding the ADHeatConductionRZ object.
+The equation term added is the same as for the [HeatConduction.md] kernel,
+except that the residual and Jacobian contributions are derived for cylindrical coordinates, the axis and origin
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/kernels/ADHeatConductionRZ.md:6]:
```suggestion
except that the residual and Jacobian contributions are integrated with cylindrical coordinates, the axis and origin
```
prefer that?

File: modules/thermal_hydraulics/src/kernels/OneD3EqnMomentumFlux.C [hunk 3]
```diff
@@ -24,7 +24,7 @@ OneD3EqnMomentumFlux::validParams()
   params.addRequiredParam<MaterialPropertyName>("rho", "Density material property");
   params.addRequiredParam<MaterialPropertyName>("vel", "Velocity material property");
   params.addRequiredParam<MaterialPropertyName>("p", "Pressure material property");
-  params.addClassDescription("Momentum flux for 1-phase flow");
+  params.addClassDescription("Computes a omentum flux term for 1-phase flow");
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/src/kernels/OneD3EqnMomentumFlux.C:27]:
```suggestion
  params.addClassDescription("Computes a momentum flux term for 1-phase flow");
```

File: modules/thermal_hydraulics/src/kernels/OneDEnergyWallHeatFlux.C [hunk 4]
```diff
@@ -17,6 +17,7 @@ OneDEnergyWallHeatFlux::validParams()
   InputParameters params = Kernel::validParams();
   params.addRequiredParam<MaterialPropertyName>("q_wall", "Wall heat flux material property");
   params.addRequiredCoupledVar("P_hf", "heat flux perimeter");
+  params.addClassDescription("Add a heat flux along the local heated perimeter");
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/src/kernels/OneDEnergyWallHeatFlux.C:20]:
```suggestion
  params.addClassDescription("Adds a heat flux along the local heated perimeter");
```

File: modules/thermal_hydraulics/src/kernels/CoupledForceRZ.C [hunk 5]
```diff
@@ -16,6 +16,8 @@ CoupledForceRZ::validParams()
 {
   InputParameters params = CoupledForce::validParams();
   params += RZSymmetry::validParams();
+  params.addClassDescription(
+      "Add a coupled force term in XY coordinates interpreted as cylindrical coordinates");
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/src/kernels/CoupledForceRZ.C:20]:
```suggestion
      "Adds a coupled force term in XY coordinates interpreted as cylindrical coordinates");
```

File: modules/thermal_hydraulics/src/kernels/ADHeatConductionRZ.C [hunk 6]
```diff
@@ -16,6 +16,8 @@ ADHeatConductionRZ::validParams()
 {
   InputParameters params = ADHeatConduction::validParams();
   params += RZSymmetry::validParams();
+  params.addClassDescription(
+      "Add a heat conduction term in XY coordinates interpreted as cylindrical coordinates");
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/src/kernels/ADHeatConductionRZ.C:20]:
```suggestion
      "Adds a heat conduction term in XY coordinates interpreted as cylindrical coordinates");
```

File: modules/thermal_hydraulics/src/kernels/ADHeatConductionTimeDerivativeRZ.C [hunk 7]
```diff
@@ -16,6 +16,8 @@ ADHeatConductionTimeDerivativeRZ::validParams()
 {
   InputParameters params = ADHeatConductionTimeDerivative::validParams();
   params += RZSymmetry::validParams();
+  params.addClassDescription("Add a time derivative term for the energy equation in XY coordinates "
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/src/kernels/ADHeatConductionTimeDerivativeRZ.C:19]:
```suggestion
  params.addClassDescription("Adds a time derivative term for the energy equation in XY coordinates "
```

File: modules/thermal_hydraulics/src/kernels/ADHeatStructureHeatSource.C [hunk 8]
```diff
@@ -15,6 +15,7 @@ InputParameters
 ADHeatStructureHeatSource::validParams()
 {
   InputParameters params = ADKernel::validParams();
+  params.addClassDescription("Add a heat source term for the energy equation");
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/src/kernels/ADHeatStructureHeatSource.C:18]:
```suggestion
  params.addClassDescription("Adds a heat source term for the energy equation");
```

File: modules/thermal_hydraulics/src/kernels/ADHeatStructureHeatSourceRZ.C [hunk 9]
```diff
@@ -16,6 +16,8 @@ ADHeatStructureHeatSourceRZ::validParams()
 {
   InputParameters params = ADHeatStructureHeatSource::validParams();
   params += RZSymmetry::validParams();
+  params.addClassDescription(
+      "Add a heat source term in XY coordinates interpreted as cylindrical coordinates");
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/src/kernels/ADHeatStructureHeatSourceRZ.C:20]:
```suggestion
      "Adds a heat source term in XY coordinates interpreted as cylindrical coordinates");
```

File: modules/thermal_hydraulics/src/scalarkernels/ADShaftTimeDerivativeScalarKernel.C [hunk 10]
```diff
@@ -19,7 +19,7 @@ ADShaftTimeDerivativeScalarKernel::validParams()
   InputParameters params = ADScalarTimeDerivative::validParams();
   params.addRequiredParam<std::vector<UserObjectName>>("uo_names",
                                                        "Names of shaft-connectable user objects");
-
+  params.addClassDescription("Add a time derivative term to the shaft ODE");
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/src/scalarkernels/ADShaftTimeDerivativeScalarKernel.C:22]:
```suggestion
  params.addClassDescription("Adds a time derivative term to the shaft ODE");
```

File: modules/thermal_hydraulics/src/scalarkernels/ShaftTimeDerivativeScalarKernel.C [hunk 11]
```diff
@@ -21,7 +21,7 @@ ShaftTimeDerivativeScalarKernel::validParams()
   InputParameters params = ScalarKernel::validParams();
   params.addRequiredParam<std::vector<UserObjectName>>("uo_names",
                                                        "Names of shaft-connectable user objects");
-
+  params.addClassDescription("Add a time derivative term to the shaft ODE");
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/src/scalarkernels/ShaftTimeDerivativeScalarKernel.C:24]:
```suggestion
  params.addClassDescription("Adds a time derivative term to the shaft ODE");
```

**Comment by GiudGiud** [comment]
should be fairly ready for review. Civet is hosed rn

File: modules/thermal_hydraulics/doc/content/source/kernels/ADHeatConductionRZ.md (hunk 2 above)
Comment (joshuahansel) [reply modules/thermal_hydraulics/doc/content/source/kernels/ADHeatConductionRZ.md:6]: Yeah

File: modules/thermal_hydraulics/doc/content/source/base/FlowModelSinglePhase.md [hunk 12]
```diff
@@ -0,0 +1,16 @@
+# FlowModelSinglePhase
+
+The `FlowModelSinglePhase` creates for [FlowChannel1Phase.md] components using the three equations model:
```
Comment (joshuahansel) [inline modules/thermal_hydraulics/doc/content/source/base/FlowModelSinglePhase.md:3]:
```suggestion
The `FlowModelSinglePhase` creates the following for [FlowChannel1Phase.md] components using the 3-equation model:
```

File: modules/thermal_hydraulics/doc/content/source/base/FlowModelSinglePhase.md [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+
+- variables
+- initial conditions
+- kernels and DGKernels for RDG
```
Comment (joshuahansel) [inline modules/thermal_hydraulics/doc/content/source/base/FlowModelSinglePhase.md:7]:
```suggestion
- kernels and DGKernels
```

**Review by joshuahansel — CHANGES_REQUESTED** [review]

File: modules/thermal_hydraulics/doc/content/source/kernels/ADHeatConductionTimeDerivativeRZ.md [hunk 14]
```diff
@@ -1,20 +1,20 @@
...
-MooseObject; however, what is contained is ultimately determined by what is necessary to make the
-documentation clear for users.
-
 !syntax description /Kernels/ADHeatConductionTimeDerivativeRZ
 
-## Overview
-
-!! Replace these lines with information regarding the ADHeatConductionTimeDerivativeRZ object.
+The equation term added is the same as for the [HeatConductionTimeDerivative.md] kernel,
+except that the residual and Jacobian contributions are derived for cylindrical coordinates, the axis and origin
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/source/kernels/ADHeatConductionTimeDerivativeRZ.md:6]:
```suggestion
except that the residual and Jacobian contributions are integrated with cylindrical coordinates, the axis and origin
```

**Comment by GiudGiud** [comment]
fake fails

**Review by joshuahansel — APPROVED** [review]

**Pull request merged**